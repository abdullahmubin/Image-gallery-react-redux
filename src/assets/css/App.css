/* gallery panel style */

.task-card{
  margin-top: 1rem;
	margin-bottom: 1rem; 
	padding: 0.75rem;
	background-color: #718096;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1), 0 1px 2px 0 rgba(0,0,0,0.06);
  width: 70%;
	margin: auto;
	border-radius: 0.25rem;
  color: #fff;
  margin-bottom: 15px;
}

.drop-item-container {
  border: #282c34 dashed;
  /* float: right; */
  width: 70%;
  min-height: 550px;
  max-height: 600px;
  overflow: auto;
  margin: auto;
}
.drop-item-container > .task-card {
  width: 200px !important;
  margin-right: 5px;
  float: left;
}

.main-container {
  width: 100%;
  height: 93vh
}

.media-panel-container{
	border-radius: 0.25rem;
	padding: 0.75rem;
	box-shadow: 0 4px 6px -1px rgba(0,0,0,0.1), 0 2px 4px -1px rgba(0,0,0,0.06);
	float: left;
  width: 300px;
  margin-top: 25px;
  max-height: 600px;
  overflow: auto;
}

.canvas-container {
	/* box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px, rgba(0, 0, 0, 0.06) 0px 1px 2px 0px;
	 background-color: rgb(154, 230, 180); 
	min-height: 200px;
	text-align: center; */
  color: rgb(255, 255, 255);
  /* border: #282c34 dashed; */
}

.container {
  width: 100px;
  max-width: 400px;
}
.overlay {
  position: relative;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  
  width: 100%;
  opacity: 0;
  transition: .3s ease;
 background-color: red;
}

.container:hover .overlay {
  opacity: 1;
}

.icon {
  
  font-size: 100px;
  position: absolute;
  top: 10%;
  left: 50%;
  transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  text-align: center;
}

.fa-user:hover {
  color: #eee;
}

.icon-holder {
  position: relative;
  right: 130px;
  height: 10px;
  color: #dddffd;
}
.remove-icon {
  position: relative;
  height: 10px;
}
.toolbar-holder{
  position: relative;
    right: 150px;
    display: none;
    width: 400px;
    float: left;

}

.icon-holder:hover {
  /* color: #81aaf3; */
}
.show-image-window{
  margin-left: 20px;
}
.show-img-name {
  margin-left: 40px;
    background: #718096;
    color: #000;
    font-weight: bold;
    position: absolute;
    display: none;
}
.img-details:hover + .show-img-name {
  display: block
}
.show-image-window:hover + .toolbar-holder {
  /* height: 20px; */
  display: block;
}

.done-img-container {
    display: flex;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}
.done-task-card-container{
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding: 0.75rem;
  background-color: #718096; 
  opacity: 1;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1), 0 1px 2px 0 rgba(0,0,0,0.06);
  width: 100%;
  border-radius: 0.25rem;
  color: #fff;
  float: left;
}
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
