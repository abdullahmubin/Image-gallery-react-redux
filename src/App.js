import './assets/css/App.css';
import GalleryInfo from './component/galleryInfo';
import { Provider } from 'react-redux';
import store from './store/store';

import { HTML5Backend } from 'react-dnd-html5-backend';
import { DndProvider } from 'react-dnd';

function App() {
  return (
    <Provider store={store}>
      <DndProvider backend={HTML5Backend}>
        <div className="App">
          <GalleryInfo />
        </div>
      </DndProvider>
    </Provider>

  );
}


export default App;