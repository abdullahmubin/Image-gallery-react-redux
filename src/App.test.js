import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from "react-test-renderer";

test('getbytext', () => {
  render(<App />);
  const linkElement = screen.getByText(/Media Panel/i);
  expect(linkElement).toBeInTheDocument();
});

if("renders without crashing", ()=> {
  const div = document.createElement("div");
  ReactDOM.render(<img />, div);
})

describe("snapshot testing", () => {
  test("renders when there are no items", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot(`<div />`);
  });
});
