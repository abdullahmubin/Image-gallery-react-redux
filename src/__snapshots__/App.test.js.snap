// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing renders when there are no items: <div /> 1`] = `
<div
  className="App"
>
  <div
    className="main-container"
  >
    <div
      className="media-panel-container"
    >
      <div>
        <p>
          Media Panel
        </p>
      </div>
    </div>
    <div
      className="canvas-container"
    >
      <div>
        <div
          className="drop-item-container"
        />
      </div>
    </div>
  </div>
</div>
`;
